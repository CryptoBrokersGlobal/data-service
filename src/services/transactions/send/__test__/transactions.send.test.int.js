const createService = require('..');
const { createPgDriver } = require('../../../../db');

// first ID has 12 txs with it
const TX_ID = '5jpwaJnERa8Gr1ChgrNYnmxm2EtZ4KHC5bW1ZLL7LCY1bUV9gFWFAGjpJaPDCawmFzguqGBgYDyeocpEsKWeYDM1';
const TX_ID_2 = '119tSRyy2sAFcAS5Htb6m4FtsCtJsacAwarg75WpsAf9HgMJnpvgcLou31ARxxEp1WMuTmGTRNUeghCgrS4PxDT';

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('Send transaction service', () => {
  describe('get', () => {
    it('fetches real tx', async done => {
      service
        .get(TX_ID)
        .run()
        .promise()
        .then(x => {
          expect(x).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
    it('returns null for unreal tx', async () => {
      const tx = await service
        .get('UNREAL')
        .run()
        .promise();

      expect(tx).toBe(null);
    });
  });

  describe('mget', () => {
    it('fetches real txs with nulls for unreal', async done => {
      service
        .mget([TX_ID, 'UNREAL', TX_ID_2])
        .run()
        .promise()
        .then(xs => {
          expect(xs).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
  });
});
