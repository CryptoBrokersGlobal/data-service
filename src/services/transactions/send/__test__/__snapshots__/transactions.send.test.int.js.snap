// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send transaction service get fetches real tx 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "amount": "5",
    "fee": "0.001",
    "height": 251375,
    "id": "5jpwaJnERa8Gr1ChgrNYnmxm2EtZ4KHC5bW1ZLL7LCY1bUV9gFWFAGjpJaPDCawmFzguqGBgYDyeocpEsKWeYDM1",
    "recipient": "3PR3xnzz1oD58K73T1dkLuVMJG8UxDZZJzV",
    "sender": "3P89Gonteikxhpup4ExAf5SaQyHVzeQqvNg",
    "senderPublicKey": "4uQeVj5tqViQh7yWWGStvkEG1Zmhx6uasJtWCJziofM",
    "signature": "5jpwaJnERa8Gr1ChgrNYnmxm2EtZ4KHC5bW1ZLL7LCY1bUV9gFWFAGjpJaPDCawmFzguqGBgYDyeocpEsKWeYDM1",
    "timestamp": 2016-11-28T09:51:07.653Z,
    "type": 2,
  },
}
`;

exports[`Send transaction service mget fetches real txs with nulls for unreal 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "5",
        "fee": "0.001",
        "height": 251375,
        "id": "5jpwaJnERa8Gr1ChgrNYnmxm2EtZ4KHC5bW1ZLL7LCY1bUV9gFWFAGjpJaPDCawmFzguqGBgYDyeocpEsKWeYDM1",
        "recipient": "3PR3xnzz1oD58K73T1dkLuVMJG8UxDZZJzV",
        "sender": "3P89Gonteikxhpup4ExAf5SaQyHVzeQqvNg",
        "senderPublicKey": "4uQeVj5tqViQh7yWWGStvkEG1Zmhx6uasJtWCJziofM",
        "signature": "5jpwaJnERa8Gr1ChgrNYnmxm2EtZ4KHC5bW1ZLL7LCY1bUV9gFWFAGjpJaPDCawmFzguqGBgYDyeocpEsKWeYDM1",
        "timestamp": 2016-11-28T09:51:07.653Z,
        "type": 2,
      },
    },
    null,
    Object {
      "__type": "transaction",
      "data": Object {
        "amount": "0.00001",
        "fee": "1e-8",
        "height": 21874,
        "id": "119tSRyy2sAFcAS5Htb6m4FtsCtJsacAwarg75WpsAf9HgMJnpvgcLou31ARxxEp1WMuTmGTRNUeghCgrS4PxDT",
        "recipient": "3PBrHA9R2Rf6xg4keyymMvrvNQmc6oQhxgD",
        "sender": "3PNqJ6aaGAL15fmPHhWvk3V6yqL6RpptMZp",
        "senderPublicKey": "HCt7iRNfcWSv4A6yPgcLaBQ6URWAa8ZdPQC9FUkbbJSW",
        "signature": "119tSRyy2sAFcAS5Htb6m4FtsCtJsacAwarg75WpsAf9HgMJnpvgcLou31ARxxEp1WMuTmGTRNUeghCgrS4PxDT",
        "timestamp": 2016-06-27T10:07:19.161Z,
        "type": 2,
      },
    },
  ],
}
`;
